src/lib/SnackBar/index.ts

```diff
@@ -0,0 +1,98 @@
+import './styles.css';
+
+const DEFAULT_TIMEOUT = 2750;
+
+export interface SnackBarOptions {
+  message: string;
+  timeout?: number;
+  actionText?: string;
+  actionHandler?: () => boolean | null;
+}
+
+class SnackBar {
+  on?: () => void;
+  onhide?: () => void;
+  private options: SnackBarOptions;
+  private element: Element = document.createElement('div');
+  private text: Element = document.createElement('div');
+  private button: Element = document.createElement('button');
+  private parent: Element;
+  private showing = false;
+  private closeTimer?: number;
+
+  constructor (options: SnackBarOptions, parent: Element) {
+    this.options = options;
+    this.parent = parent;
+
+    this.element.className = 'snackbar';
+    this.element.setAttribute('aria-live', 'assertive');
+    this.element.setAttribute('aria-atomic', 'true');
+    this.element.setAttribute('aria-hidden', 'true');
+
+    this.text.className = 'snackbar--text';
+    this.text.textContent = options.message;
+    this.element.appendChild(this.text);
+
+    if (options.actionText) {
+      this.button.className = 'snackbar--button';
+      this.button.textContent = options.actionText || '';
+      this.button.addEventListener('click', (event) => {
+        if (options.actionHandler && options.actionHandler() === false) return;
+        this.hide();
```

Comment:
Since hide is async, it's possible to click this button multiple times. Do we want that?

Reply:
There's a boolean `_showing` flag checked in hide() to prevent this. I've added the same guard to actionHandler invocation though.